lib/services/application.js

```diff
@@ -0,0 +1,107 @@
+const db = require('../storage/mongo');
+const fs = require('fs');
+const config = require('../../config')
+const formidable = require('formidable');
+
+const applicationGet =  async (name) => {
+    try {
+        if (name){
+            const app_versions = await db.getApplication({name})
+            const ret = app_versions.map(app => (app.version))
+            return ret
+        }
+    }
+    catch (error){
+        console.log("Error in applicationGet " + error.message)
+        return undefined
+    }
+}
+
+const getLatestVersion = async (name) => {
+    try {
+        if (name){
+            const latest_version = await db.getLatestVersion({name})
+            const ret = latest_version.map(app => (app.version))
+            return ret
+        }
+    }
+    catch (error){
+        console.log("Error in getLatestVersion " + error.message)
+        return undefined
+    }
+}
+
+const applicationGetFile = async (name, version) => {
+    try {
+        if(!name)
+            throw new Error("Application name must be valid")
+        if(!version)
+            throw new Error("Application version must be valid")
+        console.log("@",name, version)
+        if (name && version){
+            const application_doc = await db.getApplication({name, version})
+            const file = application_doc.map(doc => (doc.path))
+            const path = file[0]
+            return path
+        }
+        }
+    catch (error){
+        console.log("Error in getLatestVersion " + error.message)
+        return undefined
+    }
+}
+
+const applicationPost = async (req, name, version) => {
+    try{
+        if (!fs.existsSync(config.applicationConfig.filesLocation)){
+            fs.mkdirSync(config.applicationConfig.filesLocation);
+        }
+        const app_dir = config.applicationConfig.filesLocation + "/"+name
+        if (!fs.existsSync(app_dir)){
+            fs.mkdirSync(app_dir);
+        }
+
+        const ver_dir = app_dir+"/"+version
+        if (!fs.existsSync(ver_dir)){
+            fs.mkdirSync(ver_dir);
+        }
+        
+        var form = new formidable.IncomingForm();
```

Comment:
I'd rather you put the form handling in the route, and not pass the request object (which is express specific) to the service.